tensorflow-core/tensorflow-core-api/pom.xml

```diff
@@ -14,12 +14,22 @@
   <name>TensorFlow Core API Library</name>
   <description>Platform-dependent native code and pure-Java code for the TensorFlow machine intelligence library.</description>
 
+  <properties>
+    <!-- Match version used by TensorFlow, in tensorflow/workspace.bzl -->
+    <protobuf.version>3.8.0</protobuf.version>
+  </properties>
+
   <dependencies>
     <dependency>
       <groupId>org.bytedeco</groupId>
       <artifactId>javacpp</artifactId>
       <version>${javacpp.version}</version>
     </dependency>
+    <dependency>
+      <groupId>com.google.protobuf</groupId>
+      <artifactId>protobuf-java</artifactId>
+      <version>${protobuf.version}</version>
```

Comment:
Using the original protobuf artifact as is will cause issues with other libraries that need a different version of protobuf, so this should all be shaded. If the plan is to wait until users having this problem show up, is it going to be OK to break backward compatibility at that time?

Reply:
So embedding it using Maven shade plugin? I can give it a try.

Reply:
Not exactly, we need to use a patched version of protobuf that can generate code that depends on shaded classes, like this one: https://github.com/os72/protoc-jar-maven-plugin

But like I said it's not important to do that right away if we plan on breaking compatibility at some point in the future anyway.

Reply:
Oh, that might require more changes then because right now, I let [Bazel generating the Java protobuf classes](https://github.com/karllessard/tensorflow-java/blob/d7618c2b9be4b5f0d4942ef8adbd571e8aa6bc43/tensorflow-core/tensorflow-core-api/BUILD#L82) and just match the version used in Maven, as I did not find how to export only the `.proto` files out of TensorFlow.

I also found out [this](https://github.com/tensorflow/ecosystem/blob/1b3a0529bfa7e8eca6419cb5bef4d6f85b86c81e/spark/spark-tensorflow-connector/pom.xml#L109), for what it is worth. Looks like there is already some `org.tensorflow` artifacts that shades protobuf in a different way.

Reply:
Right, we can let users do it themselves when required...